tracing-mock/src/event.rs

```diff
@@ -66,18 +172,168 @@ impl ExpectedEvent {
+    ///     tracing::info!(parent: parent.id(), field = &"value");
+    /// });
+    ///
+    /// handle.assert_finished();
+    /// ```
+    pub fn with_explicit_parent(self, parent: &str) -> ExpectedEvent {
+        Self {
+            parent: Some(Parent::Explicit(parent.into())),
+            ..self
+        }
+    }
+
+    /// Adds a validation that the event has no explicit parent.
+    ///
+    /// ```
+    /// use tracing::collect::with_default;
+    /// use tracing_mock::{collector, expect};
+    ///
+    /// let event = expect::event()
+    ///     .without_explicit_parent();
+    ///
+    /// let (collector, handle) = collector::mock()
+    ///     .event(event)
+    ///     .run_with_handle();
+    ///
+    /// with_default(collector, || {
+    ///     tracing::info!(field = &"value");
+    /// });
+    ///
+    /// handle.assert_finished();
+    /// ```
+    pub fn without_explicit_parent(self) -> ExpectedEvent {
+        Self {
+            parent: Some(Parent::ExplicitRoot),
+            ..self
+        }
+    }
+
+    /// Sets the expected contextual parent to match an event.
+    ///
+    /// ```
+    /// use tracing::collect::with_default;
+    /// use tracing_mock::{collector, expect};
+    ///
+    /// let event = expect::event()
+    ///     .with_contextual_parent("parent_span");
+    ///
+    /// let (collector, handle) = collector::mock()
+    ///     .enter(expect::span())
+    ///     .event(event)
+    ///     .run_with_handle();
+    ///
+    /// with_default(collector, || {
+    ///     let parent = tracing::info_span!("parent_span");
+    ///     let _guard = parent.enter();
+    ///     tracing::info!(field = &"value");
+    /// });
+    ///
+    /// handle.assert_finished();
+    /// ```
+    pub fn with_contextual_parent(self, parent: &str) -> ExpectedEvent {
+        Self {
+            parent: Some(Parent::Contextual(parent.into())),
+            ..self
+        }
+    }
+
+    /// Adds a validation that the event has no contextual parent.
+    ///
+    /// ```
+    /// use tracing::collect::with_default;
+    /// use tracing_mock::{collector, expect};
+    ///
+    /// let event = expect::event()
+    ///     .with_contextual_parent("parent_span");
+    ///
+    /// let (collector, handle) = collector::mock()
+    ///     .enter(expect::span())
+    ///     .event(event)
+    ///     .run_with_handle();
+    ///
+    /// with_default(collector, || {
+    ///     let parent = tracing::info_span!("parent_span");
+    ///     let _guard = parent.enter();
+    ///     tracing::info!(field = &"value");
+    /// });
+    ///
+    /// handle.assert_finished();
+    /// ```
+    pub fn without_contextual_parent(self) -> ExpectedEvent {
+        Self {
+            parent: Some(Parent::ContextualRoot),
+            ..self
+        }
+    }
+
+    /// Validates that the event is emitted within the scope of the
+    /// provided `spans`.
+    ///
+    /// Note: This validation currently only works with a
```

Comment:
nit: generally we bold `Note`s in RustDoc.
```suggestion
    /// **Note**: This validation currently only works with a
```

Reply:
Change applied.